ucb-rit/coldfront
# Add password change/reset functionality

Fixes #4 

**Changes**
- Added support for password reset functionality for unauthenticated users, reachable via a "Forgot your password?" link at login, as well as via a link in a message that appears when the provided login credentials are incorrect.
- Added support for password change functionality for authenticated users, reachable via the User Profile.
- Added password validation to disallow bad passwords.

**How to Test**
- Review the code changes and add comments as needed.
- Add the following to `coldfront/config/local_settings.py` so that emails are printed to the shell:
    ```
    EMAIL_BACKEND = 'django.core.mail.backends.console.EmailBackend'
    ```
- Copy `coldfront/config/local_strings.py.sample` to `coldfront/config/local_strings.py` to apply the new login fail message.
- As an unauthenticated user, test the password reset functionality at login.
    - Inputting a valid email address should cause an email to be printed to the shell. Following the link should lead to a password reset form.
    - Test various passwords that do not conform to the guidelines and ensure that they fail.
    - Test that inputting a valid new password changes the password.
- As an authenticated user, test the password change functionality in the User Profile in a similar fashion.
